FileSystemExtensionLibrary/FileSystemVisitor.cs

```diff
@@ -1,150 +0,0 @@
-﻿using FileSystemExtensionLibrary.EventArgs;
-using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Security;
-using System.Security.Permissions;
-
-namespace FileSystemExtensionLibrary
-{
-    public sealed class FileSystemVisitor
-    {
-        #region  Fields
-        private readonly Func<FileSystemInfo, bool> filter;
-        #endregion
-
-        #region  Constructors
-        public FileSystemVisitor() { }
-
-        public FileSystemVisitor(Func<FileSystemInfo, bool> filter = null)
-        {
-            this.filter = filter;
-        }
-        #endregion
-
-        #region  Events
-        public event EventHandler<StartEventArgs> Start;
-        public event EventHandler<FinishEventArgs> Finish;
-        public event EventHandler<ItemFindedEventArgs<FileInfo>> FileFinded;
-        public event EventHandler<ItemFilteredEventArgs<FileInfo>> FilteredFileFinded;
-        public event EventHandler<ItemFindedEventArgs<DirectoryInfo>> DirectoryFinded;
-        public event EventHandler<ItemFilteredEventArgs<DirectoryInfo>> FilteredDirectoryFinded;
-        #endregion
-
-        public IEnumerable<FileSystemInfo> GetFileSystemInfoSequence(string path)
-        {
-            if (String.IsNullOrEmpty(path))
-            {
-                throw new ArgumentNullException("path");
-            }
```

Comment:
I'd rather check directory existence by  Directory.Exists(path)

Reply:
I agree with you, it is will be better.